pandas/tools/pivot.py

```diff
@@ -106,6 +106,10 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
         else:
             values_multi = False
             values = [values]
+        # GH14938 Make sure values are in data
+        for i in values:
+            if i not in data:
+                raise KeyError(i)
```

Comment:
Good point. Something like this would list them all:

```python
missing_values = set(values) - set(data)
if missing_values:
    raise KeyError('{} missing in [better message]'.format(', '.join(missing_values)
``` 

But if you have strong view on looping then OK